shared/js/background/onboarding.es6.js

```diff
@@ -0,0 +1,115 @@
+/**
+* This is only injected on duckduckgo.com to assist with user onboarding
+*/
+
+function createOnboardingCode ({
+    isAddressBarQuery,
+    showWelcomeBanner,
+    showCounterMessaging,
+    extensionId,
+    browser
+} = {}) {
+    const params = JSON.stringify({
+        isAddressBarQuery,
+        showWelcomeBanner,
+        showCounterMessaging,
+        extensionId,
+        browser
+    })
+
+    // TODO: upgrade to `chrome.scripting.executeScript` when we upgrade to manifest v3
+    // as it allows to inject a function with _arguments_. Here we simulate that in a hacky way
+    return `(${onboarding.toString()})(${params})`
+}
+
+function onboarding ({
+    isAddressBarQuery,
+    showWelcomeBanner,
+    showCounterMessaging,
+    browser,
+    extensionId
+}) {
+    function getDocumentStartData (cb) {
+        if (browser !== 'chrome') {
+            return cb(null)
+        }
+
+        window.postMessage({ type: 'documentStartDataRequest' }, window.location.origin)
+        window.addEventListener('message', function handleMessage (e) {
+            if (e.origin === window.location.origin && e.data.type === 'documentStartDataResponse') {
+                window.removeEventListener('message', handleMessage)
+                cb(null, e.data.payload)
+            }
+        })
+    }
+
+    function start () {
+        // Consolidate data: grab value of the Chrome Dialogue content-script
+        // this script was injected earlier to capture value as early as possible
+        getDocumentStartData((err, documentStartData) => {
+            if (err) {
```

Comment:
Can we remove this argument if it's not currently used?

Reply:
>> Can we remove this argument if it's not currently used?

> documentStartData is used see:

Again I meant `err` sorry for the confusion.

--- 

As for the other comment...

> Note that the url filter will make this harder etc

Essentially we have the ddg url hardcoded here is what I meant (harder was a wrong choice in words) so we still have the local development issue currently.

Ah! so the regex is only there for local development. Makes sense; we have this issue elsewhere with most things in `constants.js` I suspect we should hardcode to "https://duckduckgo.com" and also put it in that file.

This reduces the risk IMO as that regex could be wrongly interpreted and also we won't inject it into other properties like chat.duckduckgo.com etc.

Reply:
We are now using `hostEquals: constants.duckDuckGoSerpHostname` (we will improve the DX in a subsequent PR)